brian2/tests/test_GSL.py

```diff
@@ -0,0 +1,562 @@
+from nose import with_setup, SkipTest
+from nose.plugins.attrib import attr
+from numpy.testing.utils import assert_allclose, assert_equal, assert_raises
+
+from brian2 import *
+from brian2.devices.device import reinit_devices
+from brian2.core.preferences import PreferenceError
+
+from brian2.codegen.runtime.GSLcython_rt import IntegrationError
+from brian2.stateupdaters.base import UnsupportedEquationsException
+
+max_difference = .1*mV
+
+
+@attr('standalone-compatible')
+@with_setup(teardown=reinit_devices)
+def test_GSL_stateupdater_basic():
+    try:
+        # just the adaptive_threshold example: run for exponential_euler and GSL and see
+        # if results are comparable (same amount of spikes and spikes > 0)
+        eqs = '''
+        dv/dt = -v/(10*ms) : volt
+        dvt/dt = (10*mV-vt)/(15*ms) : volt
+        '''
+        reset = '''
+        v = 0*mV
+        vt += 3*mV
+        '''
+        neurons_conventional = NeuronGroup(1, model=eqs, reset=reset,
+                                           threshold='v>vt', method='exponential_euler')
+        neurons_GSL = NeuronGroup(1, model=eqs, reset=reset,
+                                  threshold='v>vt', method='gsl')
+        neurons_conventional.vt = 10*mV
+        neurons_GSL.vt = 10*mV
+        # 50 'different' neurons so no neuron spikes more than once per dt
+        P = SpikeGeneratorGroup(1, [0]*50, array(range(50))/50.*100*ms)
+        C_conventional = Synapses(P, neurons_conventional, on_pre='v += 3*mV')
+        C_GSL = Synapses(P, neurons_GSL, on_pre='v += 3*mV')
+        C_conventional.connect()
+        C_GSL.connect()
+        SM_conventional = SpikeMonitor(neurons_conventional, variables='v')
+        SM_GSL = SpikeMonitor(neurons_GSL, variables='v')
+        net = Network(neurons_conventional, neurons_GSL, P,
+                      C_conventional, C_GSL, SM_conventional, SM_GSL)
+        net.run(100*ms)
+        assert SM_conventional.num_spikes > 0, 'simulation should produce spiking, but no spikes monitored'
+        assert SM_conventional.num_spikes == SM_GSL.num_spikes, ('GSL_statupdater produced different number '
+                                                                 'of spikes than integration with ',
+                                                                 'exponential euler')
+    except NotImplementedError:
```

Comment:
I agree that this is ugly, I was hoping that it would be only temporary, but I think numpy support for GSL is still a bit off... We could do something that we do in other places:
```Python
if prefs.codegen.target == 'numpy':
    raise SkipTest('...')
```
but this is not great since we actually want to test whether the integration correctly raises a `NotImplementedError`. We could also globally accept `NotImplementedError` (we do this for `brian2genn`), but in other places I *would* consider `NotImplementedError` a test failure.
I think I'll add a small decorator to the file that does the same thing we are currently doing but looks less intrusive.